src/types/Typechecker/TypeError.hs

```diff
@@ -799,9 +821,17 @@ instance Show Error where
     show (ActiveTraitError trait) =
         printf "Trait '%s' can only be included by active classes"
                (show trait)
-    show (UnsafeTypeArgumentError ty) =
-        printf "Cannot use non-aliasable type '%s' as type argument."
-               (show ty)
+    show (UnsafeTypeArgumentError formal ty) =
+        if isModeless ty then
+          printf "Cannot use non-aliasable type '%s' as type argument" --TODO: This seems imprecise...
```

Comment:
Actually, the code here makes sense, albeit it's not very precise. This branch is for when we have a type that is indirectly `linear`, for example a tuple where one element is `linear`. It doesn't say "linear" in the error message as it gives the same error for `subord`inate types. I've kept the TODO, but elaborated on it a bit to make it more clear how the error message might be improved in the future.